cache/engine/client/client_wildcard.go

```diff
@@ -0,0 +1,125 @@
+package client
+
+//                                                                         __
+// .-----.-----.______.-----.----.-----.--.--.--.--.______.----.---.-.----|  |--.-----.
+// |  _  |  _  |______|  _  |   _|  _  |_   _|  |  |______|  __|  _  |  __|     |  -__|
+// |___  |_____|      |   __|__| |_____|__.__|___  |      |____|___._|____|__|__|_____|
+// |_____|            |__|                   |_____|
+//
+// Copyright (c) 2023 Fabio Cicerchia. https://fabiocicerchia.it. MIT License
+// Repo: https://github.com/fabiocicerchia/go-proxy-cache
+
+import (
+	"context"
+
+	goredislib "github.com/go-redis/redis/v8"
+
+	circuitbreaker "github.com/fabiocicerchia/go-proxy-cache/utils/circuit-breaker"
+)
+
+// DelWildcard - Removes the matching keys based on a pattern.
+func (rdb *RedisClient) DelWildcard(ctx context.Context, key string) (int, error) {
+	if rdb.Client.ClusterSlots(ctx).Err() == nil {
+		return rdb.deleteClusterKeys(ctx, key)
+	} else {
+		k, err := circuitbreaker.CB(rdb.Name, rdb.logger).Execute(func() (interface{}, error) {
+			keys, err := rdb.Client.Keys(ctx, key).Result()
+			return keys, err
+		})
+
+		if err != nil {
+			return 0, nil
+		}
+
+		return rdb.deleteKeys(ctx, key, k.([]string))
+	}
+}
+
+func (rdb *RedisClient) deleteClusterKeys(ctx context.Context, key string) (int, error) {
+	clusterClient := rdb.Client.(*goredislib.ClusterClient)
+	deletedKeys := Counter{}
+
+	err := clusterClient.ForEachShard(ctx, func(ctx context.Context, client *goredislib.Client) error {
+		_, err := circuitbreaker.CB(rdb.Name, rdb.logger).Execute(func() (interface{}, error) {
+			keys, err := client.Keys(ctx, key).Result()
+			if err != nil {
+				rdb.logger.Errorf("Error removing keys with pattern: %s on node: %s", key, client)
```

Comment:
is this an Error or a Warning, what's the expectation? to have the key in all the nodes?